bin/main.rb

```diff
@@ -1,2 +1,157 @@
+  puts '      .-----+-----+-----.      .-----+-----+-----.'
+  puts "      |  4  |  5  |  6  | ==>> |  #{$board[3]}  |  #{$board[4]}  |  #{$board[5]}  |"
+  puts '      .-----+-----+-----.      .-----+-----+-----.'
+  puts "      |  7  |  8  |  9  |      |  #{$board[6]}  |  #{$board[7]}  |  #{$board[8]}  |"
+  puts '      .-----.-----.-----.      .-----.-----.-----.'
+end
+
+def display_demo_board
+  puts '             .-----.-----.-----.'
+  puts '             |  X  |  O  |  X  |'
+  puts '             .-----+-----+-----.'
+  puts '             |  O  |  X  |  O  |'
+  puts '             .-----+-----+-----.'
+  puts '             |  X  |  O  |     |'
+  puts '             .-----.-----.-----.'
+end
+
+puts "\n\n"
+puts '           Welcome,  TiC TaC To\'eR!'
+puts ''
+display_demo_board
+puts ''
+puts "          Built by Robert and Grace\n\n\n"
+
+sleep(2)
+start_msg = "            Press \'S\' to start!" # Start Game
+
+2.times do
+  print "\r#{start_msg}"
+  sleep 0.5
+  print "\r#{' ' * start_msg.size}"
+  sleep 0.5
+end
+
+print "\n"
+puts "#{start_msg}.to_s"
+start_game = gets.strip.downcase
+sleep(1)
+
+puts "             OK, let\'s Play!\n\n\n" if start_game == 's'
+
+sleep(1)
+puts '           Player 1, enter your name' # Player 1
+$player_1 = gets.strip.capitalize!
+sleep(1)
+puts "              #{$player_1}, ready to go!\n\n\n"
+
+sleep(1)
+puts '           Player 2, enter your name' # Player 2
+$player_2 = gets.strip.capitalize!
+sleep(1)
+puts "              #{$player_2}, ready to go!\n\n\n"
+
+sleep(1)
+puts "            Great! #{$player_1} and #{$player_2}"
+sleep(1)
+puts "                 Let's do it!\n\n\n"
+
+sleep(2)
+puts "   Choose a cell (1-9) on the TicTacToe board!\n\n\n"
+active_board
+puts "\n\n"
+sleep(2)
+
+def player_1_turn
+  $board_input = nil?
+  $board_input = gets.chomp.to_i
+  puts "You chose cell #{$board_input}"
+  if $board[$board_input - 1] == ' '
+    $board[$board_input - 1] = 'X' # insert position
+    puts ''
+    board_choice_display
+    # check if win is 'true' return
+    $player_turn += 1
+  else
+    puts "             Cell is already chosen\n Choose an empty cell\n"
+    player_1_turn
+  end
+end
+
+def player_2_turn
+  $board_input = nil?
+  $board_input = gets.chomp.to_i
+  puts "You chose cell #{$board_input}"
+  if $board[$board_input - 1] == ' '
+    $board[$board_input - 1] = 'O'
+    puts ''
+    board_choice_display
+    # check if win is 'true' return
+    $player_turn += 1
+  else
+    puts "             Cell is already chosen\n Choose an empty cell\n"
+    player_2_turn
+  end
+end
+
+def toggle_player_turn
+  if $player_turn.odd?
+    puts "\n\n"
+    puts "              #{$player_1}, Your turn!"
+    self.player_1_turn
+  elsif $player_turn.even?
+    puts "\n\n"
+    puts "              #{$player_2}, Your turn!"
+    self.player_2_turn
+  elsif $player_turn == 10
+    print 'results'
+  end
+end
+
+while $player_turn < 10
+  toggle_player_turn
+end
+
```

Comment:
- [ ] Kindly avoid the use of definite loop such as calling a number to it OR using out-of-nowhere numbers to terminate the loop, e.g - `while $player_turn < 10`, like you have done.
Remember that the purpose of this milestone is to create a mock of the game and so, it is proper terminating the loop when some conditions are met inside the loop; like when there is a winner or the game is a draw.

HINT: Just like the first reviewer suggested, it is better to have a more descriptive variable terminating the loop. Eg: A winning_move and/or a draw_move variable as a terminating variable will do and then you can set ` $player_turn = true.`

```
while $player_turn
  ....
  if winning_move
    $player_turn = false
  end
end

```